auto_nag/scripts/variant_expiration.py

```diff
@@ -0,0 +1,301 @@
+                "f1": "short_desc",
+                "o1": "casesubstring",
+                "v1": f"The variant `{variant_id}` expiration is on",
+            },
+            bugdata=data,
+            bughandler=handler,
+        ).wait()
+
+        return data
+
+    def get_followup_action(
+        self, bug: dict, variant_id: str, bug_expiration: datetime
+    ) -> Optional[ExpirationAction]:
+        """Get the follow up action for the bug
+
+        Args:
+            bug: The bug to handle
+            variant_id: The variant id
+            bug_expiration: The expiration of the variant as appears in the bug
+        """
+        variant = self.variants.get(variant_id)
+        if variant is None:
+            return ExpirationAction.CLOSE_DROPPED
+
+        assert "bug_id" not in variant, "Variant should not be linked to multiple bugs"
+        variant["bug_id"] = bug["id"]
+
+        if variant["expiration"] > bug_expiration:
+            return ExpirationAction.CLOSE_EXTENDED
+
+        if variant["expiration"] < bug_expiration:
+            raise Exception("Variant expiration should not be decreased")
+
+        if variant["expiration"] <= self.reminder_date:
+            if not self.is_needinfoed(bug):
+                return ExpirationAction.NEEDINFO_TRIAGER
+
+            return ExpirationAction.SEND_REMINDER
+
+        return None
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+
+        summary_match = VARIANT_BUG_PAT.match(bug["summary"])
+        assert summary_match, f"Bug {bugid} has invalid summary: {bug['summary']}"
+        variant_id, bug_expiration = summary_match.groups()
+        bug_expiration = lmdutils.get_date_ymd(bug_expiration)
+
+        action = self.get_followup_action(bug, variant_id, bug_expiration)
+        if not action:
+            return None
+
+        data[bugid] = {
+            "action": action,
+            "variant_id": variant_id,
+            "expiration": bug_expiration.strftime("%Y-%m-%d"),
+        }
+
+        if action == ExpirationAction.CLOSE_DROPPED:
+            self.autofix_changes[bugid] = {
+                "status": "RESOLVED",
+                "resolution": "FIXED",
+                "comment": {
+                    "body": f"The variant has been removed from the [variants.yml]({VARIANTS_SEARCHFOX_URL}) file."
+                },
+            }
+        elif action == ExpirationAction.CLOSE_EXTENDED:
+            new_date = self.variants[variant_id]["expiration"].strftime("%Y-%m-%d")
+            self.autofix_changes[bugid] = {
+                "status": "RESOLVED",
+                "resolution": "FIXED",
+                "comment": {
+                    "body": f"Expiration date is extended to {new_date}",
+                },
+            }
+        elif action == ExpirationAction.NEEDINFO_TRIAGER:
+            if not self.add_auto_ni(bugid, utils.get_mail_to_ni(bug)):
+                data[bugid]["action"] = ExpirationAction.SKIP
```

Comment:
if the needinfo fails (say for reason of blocked due to PTO), do we warn?  can we have a human find someone else to NI?